src/transform/transform_utils/ingestion_s3_handler.py

```diff
@@ -74,10 +74,19 @@ def get_data_from_ingestion(self):
             ]
             result = {}
 
-            for table_name in table_names:
-                file_name = self.get_file_name(table_name, last_timestamp)
-                file_data_json = self.get_table_content(file_name)
+        for table_name in table_names:
+            file_name = self.get_file_name(table_name, last_timestamp)
+            file_data_json = self.get_table_content(file_name)
+
+            if file_data_json is None:
+                print(f"No data found for {table_name}")
+                continue 
+            try:
                 file_data = json.loads(file_data_json)
                 result[table_name] = file_data
+            except json.JSONDecodeError as e:
+                print(f"Error decoding JSON for table {table_name}: {e}")
+            except Exception as e:
+                print(f"Unexpected error for table {table_name}: {e}")
```

Comment:
Your logic here is good, You can keep it as it is or here is my version of solving it in a different way. Choose which one is more readable in Your opinion.

```suggestion
            if file_data_json:
                  try:
                      file_data = json.loads(file_data_json)
                      result[table_name] = file_data
                  except json.JSONDecodeError as e:
                      print(f"Error decoding JSON for table {table_name}: {e}")
                  except Exception as e:
                      print(f"Unexpected error for table {table_name}: {e}")
              else:
                    print(f"No data found for {table_name}")
```